dbms/src/Flash/Coprocessor/DAGExpressionAnalyzer.cpp

```diff
@@ -627,38 +782,13 @@ void DAGExpressionAnalyzer::appendWindowColumns(WindowDescription & window_descr
 WindowDescription DAGExpressionAnalyzer::buildWindowDescription(const tipb::Window & window)
 {
     ExpressionActionsChain chain;
-    ExpressionActionsChain::Step & step = initAndGetLastStep(chain);
-    appendSourceColumnsToRequireOutput(step);
+    ExpressionActionsChain::Step & step = createStep(this, chain);
```

Comment:
> It doesn't feel necessary to wrap a new function `createStep`.

Wrapping a function is just to make reader could quickly get the meaning of these codes, as they can know the meaning by the function name instead of comments.

Reply:
It looks like there is no difference in readability between
```
    auto & step = initAndGetLastStep(chain);
    appendSourceColumnsToRequireOutput(step);
```
and
```
    ExpressionActionsChain::Step & step = createStep(this, chain);
```
...
Instead, the latter hides the logic of `appendSourceColumnsToRequireOutput` on the function name. .

Reply:
> createStep

changed